packages/shylu/shylu_node/basker/src/shylubasker_def.hpp

```diff
@@ -664,24 +1104,940 @@ namespace BaskerNS
+        }
+
+        // ----------------------------------------------------------------------------------------------
+        // recompute MWM and AMD on each block of C
+        INT_1DARRAY blk_nnz;
+        INT_1DARRAY blk_work;
+        for (Int i = b_first; i <= btf_nblks; i++) {
+          //printf( " > btf_tabs(%d) = %d -> %d\n",i,btf_tabs(i),btf_tabs(i)-nfirst );
+          btf_tabs(i) -= nfirst;
+        }
+        btf_blk_mwm_amd(b_first, btf_nblks-b_first, BTF_C,
+                        order_blk_mwm_c, order_blk_amd_c,
+                        blk_nnz, blk_work);
+        if (Options.static_delayed_pivot != 0) {
+          // revert MWM option
+          Options.blk_matching = blk_matching_A;
+        }
+        #if 0 //debug
+        printf( " >> debug: set order_blk_mwm/amd on BTF_C to identity <<\n" );
+        for (Int i = 0; i < (Int)BTF_C.nrow; i++) {
+          order_blk_mwm_c(i) = i;
+          order_blk_amd_c(i) = i;
+        }
+        #endif
+        //for (Int i = 0; i < (Int)BTF_C.nrow; i++) printf( " x mwm_c(%d)=%d amd_c(%d)=%d\n",i,order_blk_mwm_c(i), i,order_blk_amd_c(i));
+
+        // Apply MWM perm to rows
+        permute_row(BTF_C, order_blk_mwm_c);
+
+        /*printf(" pC = [\n" );
+        for(Int j = 0; j < BTF_C.ncol; j++) {
+          for(Int k = BTF_C.col_ptr[j]; k < BTF_C.col_ptr[j+1]; k++) {
+            printf("%d %d %.16e %d\n", BTF_C.row_idx[k], j, BTF_C.val[k],k);
+          }
+        }
+        printf("];\n");*/
+
+        // Apply AMD perm to rows and cols
+        // NOTE: no need to udpate vals_order_blk_amd_array (it will read in A without permutations, and compute perm here)
+        permute_row(BTF_C, order_blk_amd_c);
+        permute_col(BTF_C, order_blk_amd_c);
+        if (BTF_E.ncol > 0) {
+          // Apply AMD perm to cols
+          permute_col(BTF_E, order_blk_amd_c, BTF_A.ncol);
+        }
+        if (btf_tabs_offset > 0) {
+          // Apply AMD perm to cols
+          permute_col(BTF_B, order_blk_amd_c);
+        }
+
+        /*printf(" ppC = [\n" );
+        for(Int j = 0; j < BTF_C.ncol; j++) {
+          for(Int k = BTF_C.col_ptr[j]; k < BTF_C.col_ptr[j+1]; k++) {
+            printf("%d %d %.16e %d\n", BTF_C.row_idx[k], j, BTF_C.val[k],k);
+          }
+        }
+        printf("];\n");*/
+
+        // shift back perm to global index
+        for (Int i = b_first; i <= btf_nblks; i++) {
+          btf_tabs(i) += nfirst;
+        }
+        for (Int i = 0; i < (Int)BTF_C.ncol; i++) {
+          order_blk_mwm_c(i) += nfirst;
+          order_blk_amd_c(i) += nfirst;
+        }
+
+        if(Options.verbose == BASKER_TRUE) {
+          std::cout << std::endl << std::endl;
+          std::cout << "Basker Factor: Time to compute and apply MWM+AMD on diagonal blocks: " << mwm_amd_perm_timer.seconds() << std::endl;
+        }
+      }
+      //for (Int i = 0; i < (Int)ncol; i++) {
```

Comment:
For a later cleanup follow up, we may want to add a macro that can be enabled to guard for this type output